lib/src/lints/avoid_debug_print_in_release/avoid_debug_print_in_release_rule.dart

```diff
@@ -128,4 +143,63 @@ class AvoidDebugPrint extends SolidLintRule {
       reporter: reporter,
     );
   }
+
+  bool _isDebugPrintNode(Identifier node) {
+    const printPath = 'package:flutter/src/foundation/print.dart';
+    const debugPrint = 'debugPrint';
+
+    final String name;
+    final String sourcePath;
+    switch (node) {
+      case PrefixedIdentifier():
+        final prefix = node.prefix.name;
+        name = node.name.replaceAll('$prefix.', '');
+        sourcePath = node.staticElement?.librarySource?.uri.toString() ?? '';
+
+      case SimpleIdentifier():
+        name = node.name;
+        sourcePath = node.staticElement?.librarySource?.uri.toString() ?? '';
+
+      default:
+        return false;
+    }
+
+    return name == debugPrint && sourcePath == printPath;
+  }
+
+  bool _isDebugCheck(Expression node) {
+    if (node.childEntities.toList()
+        case [
+          final Token token,
+          final Identifier identifier,
+        ]) {
+      return token.type == TokenType.BANG &&
+          _isReleaseModeIdentifier(identifier);
+    }
+
+    return false;
+  }
+
+  bool _isReleaseModeIdentifier(Identifier node) {
+    const String expectedPath = 'package:flutter/src/foundation/constants.dart';
+    const String kReleaseModeName = 'kReleaseMode';
```

Comment:
Wouldn't it be great to move it to class level?

Reply:
They're only used locally though. Still move?